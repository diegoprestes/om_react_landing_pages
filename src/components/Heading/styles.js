import styled, { css } from 'styled-components';

const mediaFont = (theme) => css`
  @media ${theme.media.sm} {
    font-size: ${theme.fonts.sizes['4xl']};
  }
`;

const titleSize = {
  sm: (theme) => css`
    font-size: ${theme.fonts.sizes.md};
  `,
  md: (theme) => css`
    font-size: ${theme.fonts.sizes.lg};
  `,
  lg: (theme) => css`
    font-size: ${theme.fonts.sizes.xl};
  `,
  xl: (theme) => css`
    font-size: ${theme.fonts.sizes.lg};
    ${mediaFont(theme)};
  `
};

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)};
    ${titleCase(uppercase)}
  `}
`;
