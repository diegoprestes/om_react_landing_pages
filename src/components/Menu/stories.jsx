import Menu from '.';

import links from '../NavLinks/mock';

export default {
  title: 'Menu',
  component: Menu,
  args: {
    links,
    logoData: {
      text: 'LOGO',
      link: '#target',
      srcImg: ''
    }
  },
  argTypes: {}
};

export const Template = (args) => {
  return (
    <div>
      <Menu {...args} />
    </div>
  );
};
