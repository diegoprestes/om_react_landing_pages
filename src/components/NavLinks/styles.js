import styled, { css } from 'styled-components';

export const Container = styled.nav`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column nowrap;
    align-items: center;

    @media ${theme.media.sm} {
      flex-flow: row wrap;
    }
  `}
`;
