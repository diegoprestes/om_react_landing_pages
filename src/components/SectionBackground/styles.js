import styled, { css } from 'styled-components';

const containerBackgroundActivate = (theme) => css`
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.white};
`;

export const Container = styled.div`
  ${({ theme, hasBackground }) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.primaryColor};
    ${hasBackground && containerBackgroundActivate(theme)}
  `}
`;
