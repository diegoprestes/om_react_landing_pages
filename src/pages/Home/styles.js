import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme, background }) => css`
    background: ${background || theme.colors.primaryColor};
  `}
  color: white;
`;
