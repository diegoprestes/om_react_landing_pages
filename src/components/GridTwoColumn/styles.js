import styled, { css } from 'styled-components';

import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr;
    align-items: center;
    gap: ${theme.spacings.lg};

    @media ${theme.media.sm} {
      grid-template-columns: 1fr 2fr;
    }

    ${Title} {
      margin-bottom: ${theme.spacings.xl};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacings.lg};

    @media ${theme.media.sm} {
      margin-bottom: 0;
    }
  `}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css``}
`;

export const Image = styled.img`
  ${({ theme }) => css``}
`;
