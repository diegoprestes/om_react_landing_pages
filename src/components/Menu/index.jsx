import PropTypes from 'prop-types';
import * as Styled from './styles';
import SectionContainer from '../SectionContainer';
import NavLinks from '../NavLinks';
import LogoLink from '../LogoLink';
import { Menu as IconMenu } from '@styled-icons/material-outlined/Menu';
import { Close as IconClose } from '@styled-icons/material-outlined/Close';
import { useState } from 'react';

const Menu = ({ links, logoData }) => {
  const [visible, setVisible] = useState(false);
  return (
    <>
      <Styled.Button onClick={() => setVisible(!visible)} aria-label="Open/Close menu">
        {visible ? <IconClose aria-label="Close menu" /> : <IconMenu aria-label="Open menu" />}
      </Styled.Button>
      <Styled.Container visible={visible} onClick={() => setVisible(!visible)}>
        <SectionContainer>
          <Styled.MenuContainer>
            <LogoLink {...logoData} />
            <NavLinks links={links} />
          </Styled.MenuContainer>
        </SectionContainer>
      </Styled.Container>
    </>
  );
};

Menu.propTypes = {
  ...NavLinks.propTypes,
  logoData: PropTypes.shape(LogoLink.propTypes).isRequired
};

export default Menu;
