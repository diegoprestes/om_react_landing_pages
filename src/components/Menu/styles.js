import styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';
import { Container as SectionBackground } from '../SectionBackground/styles';
import { Title as Heading } from '../Heading/styles';

const menuVisible = (theme) => css`
  visibility: visible;
  opacity: 1;
`;

export const Container = styled.div`
  ${({ theme, visible }) => css`
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    z-index: 5;
    background: ${theme.colors.white};
    height: 100vh;
    visibility: hidden;
    opacity: 0;
    transition: all 300ms ease-in-out;
    ${visible && menuVisible(theme)}

    > ${SectionContainer} {
      display: grid;
      grid-template-columns: 1fr;
      grid-template-rows: 1fr;
      height: 100vh;
      overflow-y: auto;
      align-items: center;
      padding: 0 ${theme.spacings.lg};
    }

    > ${SectionBackground} {
      min-height: 100vh;
      display: flex;
      align-items: center;
    }

    & ${Heading} {
      display: flex;
      justify-content: center;
      overflow-y: none;
      padding-bottom: ${theme.spacings.lg};
      margin-top: 0;
      margin-bottom: 0;
    }

    @media ${theme.media.sm} {
      position: static;
      height: auto;
      border-bottom: 1px solid ${theme.colors.mediumGray};
      ${menuVisible(theme)};

      > ${SectionContainer} {
        display: block;
        height: auto;
        padding: 0;
      }

      & ${Heading} {
        display: block;
        padding: 0;
      }
    }
  `}
`;

export const MenuContainer = styled.div`
  ${({ theme }) => css`
    text-align: center;
    padding: ${theme.spacings.xl} 0;

    @media ${theme.media.sm} {
      display: flex;
      justify-content: space-between;
      align-items: center;
      padding: 0;
    }
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    display: block;
    z-index: 6;
    position: fixed;
    top: 2rem;
    right: 2rem;
    width: 4rem;
    height: 4rem;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    border: none;
    cursor: pointer;

    > svg {
      width: 2.5rem;
      height: 2.5rem;
    }

    @media ${theme.media.sm} {
      display: none;
    }
  `}
`;
