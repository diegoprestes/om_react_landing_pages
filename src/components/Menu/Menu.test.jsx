import Menu from '.';
import { fireEvent, screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { theme } from '../../styles/theme';

import links from '../NavLinks/mock';
const logoData = {
  text: 'Logo',
  link: '#target'
};

describe('<Menu />', () => {
  it('should render Logo and Menu Nav', () => {
    const { container } = renderTheme(<Menu links={links} logoData={logoData} />);

    expect(screen.getByRole('heading', { hidden: true })).toBeInTheDocument();
    expect(
      screen.getByRole('navigation', { label: 'Main menu', hidden: true })
    ).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render menu mobile and button for open and close the menu', () => {
    renderTheme(<Menu links={links} logoData={logoData} />);

    const button = screen.getByLabelText('Open/Close menu');
    const menuContainer = button.nextSibling;

    expect(button).toHaveStyleRule('display', 'block');
    expect(button).toHaveStyleRule('display', 'none', {
      media: theme.media.sm
    });

    expect(menuContainer).toHaveStyleRule('opacity', '0');
    expect(screen.getByLabelText('Open menu')).toBeInTheDocument();

    fireEvent.click(button);
    expect(menuContainer).toHaveStyleRule('opacity', '1');
    expect(screen.getByLabelText('Close menu')).toBeInTheDocument();

    fireEvent.click(button);
    expect(menuContainer).toHaveStyleRule('opacity', '0');
    expect(screen.getByLabelText('Open menu')).toBeInTheDocument();
  });

  it('should not render links', () => {
    const { container } = renderTheme(<Menu logoData={logoData} />);

    expect(
      screen.queryByRole('navigation', { label: 'Main menu', hidden: true }).firstChild
    ).not.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
