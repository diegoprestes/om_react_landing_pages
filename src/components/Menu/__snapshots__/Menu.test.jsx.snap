// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should not render links 1`] = `
.c4 {
  max-width: 120rem;
  margin: 0 auto;
}

.c7 {
  color: #0A1128;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c2 {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 5;
  background: #FFFFFF;
  height: 100vh;
  visibility: hidden;
  opacity: 0;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c2 > .c3 {
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 1fr;
  height: 100vh;
  overflow-y: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 3.2rem;
}

.c2 > .sc-gtsrHT {
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 .c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow-y: none;
  padding-bottom: 3.2rem;
  margin-top: 0;
  margin-bottom: 0;
}

.c5 {
  text-align: center;
  padding: 4.0rem 0;
}

.c0 {
  display: block;
  z-index: 6;
  position: fixed;
  top: 2rem;
  right: 2rem;
  width: 4rem;
  height: 4rem;
  background: #0A1128;
  color: #FFFFFF;
  border: none;
  cursor: pointer;
}

.c0 > svg {
  width: 2.5rem;
  height: 2.5rem;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c8 > img {
  max-height: 3rem;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

@media (min-width:768px) {
  .c2 {
    position: static;
    height: auto;
    border-bottom: 1px solid #DDDDDD;
    visibility: visible;
    opacity: 1;
  }

  .c2 > .c3 {
    display: block;
    height: auto;
    padding: 0;
  }

  .c2 .c6 {
    display: block;
    padding: 0;
  }
}

@media (min-width:768px) {
  .c5 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    padding: 0;
  }
}

@media (min-width:768px) {
  .c0 {
    display: none;
  }
}

@media (min-width:768px) {
  .c9 {
    -webkit-flex-flow: row wrap;
    -ms-flex-flow: row wrap;
    flex-flow: row wrap;
  }
}

<div>
  <button
    aria-label="Open/Close menu"
    class="c0"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <div
    class="c2"
  >
    <div
      class="c3 c4"
    >
      <div
        class="c5"
      >
        <h1
          class="c6 c7"
        >
          <a
            class="c8"
            href="#target"
          >
            Logo
          </a>
        </h1>
        <nav
          aria-label="Main menu"
          class="c9"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Menu /> should render Logo and Menu Nav 1`] = `
.c4 {
  max-width: 120rem;
  margin: 0 auto;
}

.c7 {
  color: #0A1128;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c2 {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 5;
  background: #FFFFFF;
  height: 100vh;
  visibility: hidden;
  opacity: 0;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c2 > .c3 {
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 1fr;
  height: 100vh;
  overflow-y: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 3.2rem;
}

.c2 > .sc-gtsrHT {
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 .c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow-y: none;
  padding-bottom: 3.2rem;
  margin-top: 0;
  margin-bottom: 0;
}

.c5 {
  text-align: center;
  padding: 4.0rem 0;
}

.c0 {
  display: block;
  z-index: 6;
  position: fixed;
  top: 2rem;
  right: 2rem;
  width: 4rem;
  height: 4rem;
  background: #0A1128;
  color: #FFFFFF;
  border: none;
  cursor: pointer;
}

.c0 > svg {
  width: 2.5rem;
  height: 2.5rem;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c10 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0A1128;
  position: relative;
}

.c10::after {
  content: '';
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0%;
  height: 0.2rem;
  background: #DC143C;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c10:hover::after {
  left: 25%;
  width: 50%;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c8 > img {
  max-height: 3rem;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

@media (min-width:768px) {
  .c2 {
    position: static;
    height: auto;
    border-bottom: 1px solid #DDDDDD;
    visibility: visible;
    opacity: 1;
  }

  .c2 > .c3 {
    display: block;
    height: auto;
    padding: 0;
  }

  .c2 .c6 {
    display: block;
    padding: 0;
  }
}

@media (min-width:768px) {
  .c5 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    padding: 0;
  }
}

@media (min-width:768px) {
  .c0 {
    display: none;
  }
}

@media (min-width:768px) {
  .c9 {
    -webkit-flex-flow: row wrap;
    -ms-flex-flow: row wrap;
    flex-flow: row wrap;
  }
}

<div>
  <button
    aria-label="Open/Close menu"
    class="c0"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <div
    class="c2"
  >
    <div
      class="c3 c4"
    >
      <div
        class="c5"
      >
        <h1
          class="c6 c7"
        >
          <a
            class="c8"
            href="#target"
          >
            Logo
          </a>
        </h1>
        <nav
          aria-label="Main menu"
          class="c9"
        >
          <a
            class="c10"
            href="#target1"
            target="_self"
          >
            Link 1
          </a>
          <a
            class="c10"
            href="#target2"
            target="_self"
          >
            Link 2
          </a>
          <a
            class="c10"
            href="#target3"
            target="_self"
          >
            Link 3
          </a>
          <a
            class="c10"
            href="#target4"
            target="_self"
          >
            Link 4
          </a>
          <a
            class="c10"
            href="#target5"
            target="_self"
          >
            Link 5
          </a>
          <a
            class="c10"
            href="#target6"
            target="_self"
          >
            Link 6
          </a>
          <a
            class="c10"
            href="#target7"
            target="_self"
          >
            Link 7
          </a>
          <a
            class="c10"
            href="#target8"
            target="_self"
          >
            Link 8
          </a>
        </nav>
      </div>
    </div>
  </div>
</div>
`;
